test/integration/puppetlabs/services/master/master_service_test.clj

```diff
@@ -349,3 +349,82 @@
                     @reported-metrics-atom
                     :puppetserver
                     "puppetlabs.localhost.compiler.compile.production.mean")))))))))
+
+(deftest jvm-metrics-sent-to-graphite-test
+  (let [reported-metrics-atom (atom {})]
+    (with-redefs [metrics-core/build-graphite-sender
+                  (fn [_ domain]
+                    (metrics-testutils/make-graphite-sender reported-metrics-atom domain))]
+      (logutils/with-test-logging
```

Comment:
Do you know if the `with-test-logging` wrapper here is essential?  Would be nice to be able to run these tests without it if we can - to avoid hiding errors in the output that might be useful to see.  If it's going to be unnecessarily spammy if we take it out, though, I'm good with this.